lib/util/LocationStatusManager.js

```diff
@@ -0,0 +1,583 @@
+        this._locationStatusColl.updateOne(
+            { _id: location },
+            {
+                $set: {
+                    _id: location,
+                    value: this._locationStatusStore[location].getValue(),
+                }
+            },
+            { upsert: false }, err => {
+                if (err) {
+                    this._logger.error('Could not update location service status in MongoDB', {
+                        method: 'LocationStatusManager._updateServiceStatusForLocation',
+                        error: err.message,
+                    });
+                    return cb(err);
+                }
+                return cb();
+            });
+    }
+
+    /**
+     * Helper method to get zookeeper state details for given location(s)
+     * @param {string} service service name, can be one of "lifecycle", "crr", "ingestion"
+     * @param {string[]} locations location names to get
+     * @param {Function} cb callback(error, stateBySite)
+     * @return {undefined}
+     */
+    _getZkStateDetails(service, locations, cb) {
+        const stateBySite = {};
+        async.each(locations, (location, next) => {
+            const zkNamespace = this._zkDetails[service].namespace;
+            const zkStatePath = this._zkDetails[service].statePath;
+            const path = `${zkNamespace}${zkStatePath}/${location}`;
+            this._zkClient.getData(path, (err, data) => {
+                if (err) {
+                    return next(err);
+                }
+                try {
+                    const d = JSON.parse(data.toString());
+                    stateBySite[location] = d;
+                } catch (e) {
+                    return next(e);
+                }
+                return next();
+            });
+        }, error => {
+            if (error) {
+                let errMessage = 'error getting state node details';
+                if (error.name === 'NO_NODE') {
+                    errMessage = 'zookeeper path was not created on queue ' +
+                        'processor start up';
+                }
+                this._logger.error(errMessage, {
+                    method: 'LocationStatusManager._getZkStateDetails',
+                    error,
+                });
+                return cb(errors.InternalError);
+            }
+            return cb(null, stateBySite);
+        });
+    }
+
+    /**
+     * Calls the correct location state getter based
+     * on where the service state is stored
+     * @param {string} service service to pause for locations
+     * @param {string[]} locations location names to pause
+     * @param {Function} cb callback
+     * @returns {undefined}
+     */
+    _getStateDetails(service, locations, cb) {
+        if (this._mongoServices.includes(service)) {
+            const selectedLocations = {};
+            locations.forEach(loc => {
+                selectedLocations[loc] = this._locationStatusStore[loc];
+            });
```

Comment:
this should do the mapping done in the later functions (`.paused = .getServicePauseStatus(service)`), to complete the abstraction :

```suggestion
            locations.forEach(loc => {
                selectedLocations[loc] = this._locationStatusStore[loc].getValue()[service]
            });
```

(or better introduce a `LocationStatus.getService()` function 😉)